R/qc_cvs.R

```diff
@@ -0,0 +1,108 @@
+#' Check CV distribution
+#'
+#' Calculates and plots the coefficients of variation for the selected grouping.
+#'
+#' @param data A dataframe containing at least peptide, precursor or protein identifiers, information on conditions and intensity values for each peptide, precursor or protein.
+#' @param grouping The column in the input dataframe containing the grouping variables (e.g. peptides, precursors or proteins).
+#' @param condition The column in the data dataframe containing condition information (e.g. "treated" and "control").
+#' @param intensity Column containing the corresponding intensity values for each peptide or precursor.
+#' @param plot A logical indicating whether the result should be plotted. Default is TRUE.
+#' @param plot_style Variable indicating the plotting style. \code{plot_style = "boxplot"} plots a boxplot, whereas \code{plot_style = "density"} plots the CV density distribution. Default is \code{plot_style = "density"}.
+#'
+#' @return Either the median CVs in % or a plot showing the distribution of the CVs.
+#' @import dplyr
+#' @import ggplot2
+#' @importFrom tidyr drop_na
+#' @importFrom magrittr %>%
+#' @importFrom rlang .data
+#' @importFrom stats sd
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' qc_cvs(
+#' data,
+#' grouping = pep_stripped_sequence,
+#' condition = condition,
+#' intensity = fg_quantity,
+#' plot = TRUE,
+#' plot_style = "density"
+#' )
+#' }
+qc_cvs <-
+  function(data, grouping, condition, intensity, plot = TRUE, plot_style = "density")
+  {
+    if(plot == FALSE)
+    {
+      result <- data %>%
+        tidyr::drop_na({{intensity}}) %>%
+        dplyr::group_by({{condition}}, {{grouping}}) %>%
+        dplyr::mutate(cv = (stats::sd({{intensity}}) / mean({{intensity}})) * 100) %>%
+        dplyr::group_by({{condition}}) %>%
+        dplyr::summarise(median_cv = stats::median(.data$cv))
+
+      return(result)
+    }
+
+    if(plot == TRUE)
+    {
+      result <- data %>%
+      dplyr::distinct({{condition}}, {{grouping}}, {{intensity}}) %>%
+      dplyr::group_by({{condition}}, {{grouping}}) %>%
+      dplyr::mutate(cv = (stats::sd({{intensity}}) / mean({{intensity}})) * 100)
```

Comment:
Same code as above should be used, but instead of summarise you should mutate the median into another column. This can be later used in the plot. 
I think instead of copy pasting code that is almost the same here and up there, you can just do most of the computation up there and then only do the summarise for plot == FALSE and only the mutate for plot == TRUE. 
So the main part of the computation should be outside any if statement.